src/foam/lib/json/StringParser.java

```diff
@@ -9,68 +9,19 @@
 import foam.lib.parse.PStream;
 import foam.lib.parse.Parser;
 import foam.lib.parse.ParserContext;
+import foam.lib.parse.Alt;
 
 public class StringParser
-  implements Parser
+  implements Parser 
 {
-  protected final static char ESCAPE = '\\';
-  protected final static ThreadLocal<Parser> unicodeParser = ThreadLocal.withInitial(UnicodeParser::new);
-  protected final static ThreadLocal<Parser> asciiEscapeParser = ThreadLocal.withInitial(ASCIIEscapeParser::new);
-  protected final static ThreadLocal<StringBuilder> sb = new ThreadLocal<StringBuilder>() {
-
-    @Override
-    protected StringBuilder initialValue() {
-      return new StringBuilder();
-    }
-
-    @Override
-    public StringBuilder get() {
-      StringBuilder b = super.get();
-      b.setLength(0);
-      return b;
-    }
-  };
+  MultiLineStringParser mp = new MultiLineStringParser();
+  SingleLineStringParser sp = new SingleLineStringParser();
+  Alt altp = new Alt(new MultiLineStringParser(), new SingleLineStringParser());
```

Comment:
Typically what we do is extend ProxyParser and in the constructor do `super(new Alt(...))`

Let's change to that so its consistent.